## PA-01. Load Annotation Data 

Load the prepared Korean sentiment dataset for pilot annotation.
The pilot annotation will be used to test the annotation guideline before full-scale labelling.

In [3]:
import pandas as pd
from pathlib import Path

In [6]:
# Define file path
data_path = Path("../data/annotation/nsmc_ko_annotation_300.csv")

# Load Korean pilot annotation dataset
df_ko = pd.read_csv(data_path)

# Preview data
df_ko.head()

,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,8363132,NaN,진짜 괜찮은데.. 휴 가슴이 답답해질정도로 많은 생각하게 되네요,ko,movie,NaN,NaN,NaN
1,9473399,NaN,BLACK ... 불가능이란건없네요,ko,movie,NaN,NaN,NaN
2,5030353,NaN,갠적으론 8점정도.. 평점이 넘낮아 10점.,ko,movie,NaN,NaN,NaN
3,315189,NaN,호러다큐멘터리라고 잔뜩기대했지만. 호러가 아니었다;,ko,movie,NaN,NaN,NaN
4,4910465,NaN,원작 만화의 원래 제목이 나오코임. 한국에선 스타트로 출간.,ko,movie,NaN,NaN,NaN


## PA-02. Inspect Dataset

Inspect the structure and completeness of the Korean pilot annotation dataset before labeling.

In [8]:
print("Dataset shape:", df_ko.shape)
print("\nColumns:")
print(df_ko.columns.tolist())

print("\nMissing values:")
print(df_ko.isna().sum())

Dataset shape: (300, 8)

Columns:
['review_id', 'review_title', 'review_body', 'language', 'product_category', 'sentiment_label', 'confidence', 'notes']

Missing values:
review_id             0
review_title        300
review_body           0
language              0
product_category      0
sentiment_label     300
confidence          300
notes               300
dtype: int64


In [9]:
df_ko.info()

<class 'pandas.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   review_id         300 non-null    int64  
 1   review_title      0 non-null      float64
 2   review_body       300 non-null    str    
 3   language          300 non-null    str    
 4   product_category  300 non-null    str    
 5   sentiment_label   0 non-null      float64
 6   confidence        0 non-null      float64
 7   notes             0 non-null      float64
dtypes: float64(4), int64(1), str(3)
memory usage: 45.0 KB


## PA-03. Create Pilot Sample

Select a subset of 30 Korean reviews for pilot annotation.

The pilot sample is used to test label definitions, identify ambiguous cases, 
and evaluate whether the annotation guideline requires revision before full-scale annotation.

In [10]:
# Create a reproducible pilot sample
pilot_ko = df_ko.sample(n=30, random_state=42).copy()

# Reset index
pilot_ko = pilot_ko.reset_index(drop=True)

# Preview sample
pilot_ko.head()

,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,10122209,NaN,미국이 진짜 선진국인 이유가 있구나 진짜 멋지드라...최고다,ko,movie,NaN,NaN,NaN
1,5849027,NaN,최고의 명작이라고 생각합니다.아직 못보신 분들은 꼭 감상하세요.,ko,movie,NaN,NaN,NaN
2,9643272,NaN,첨부터 끝까지 재밌음!!,ko,movie,NaN,NaN,NaN
3,1404443,NaN,먹먹한 상처와 조용히 다가오는 치유,ko,movie,NaN,NaN,NaN
4,10190504,NaN,내가본영화가 뜻밖의여정이라니라 다섯군대전투네 ㅋㅋ 뜻밖의여정도봐야겟음.. 진짜최고 ...,ko,movie,NaN,NaN,NaN


In [11]:
print("Pilot sample shape", pilot_ko.shape)

pilot_ko[[
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]].head(10)

Pilot sample shape (30, 8)


,review_id,review_body,sentiment_label,confidence,notes
0,10122209,미국이 진짜 선진국인 이유가 있구나 진짜 멋지드라...최고다,NaN,NaN,NaN
1,5849027,최고의 명작이라고 생각합니다.아직 못보신 분들은 꼭 감상하세요.,NaN,NaN,NaN
2,9643272,첨부터 끝까지 재밌음!!,NaN,NaN,NaN
3,1404443,먹먹한 상처와 조용히 다가오는 치유,NaN,NaN,NaN
4,10190504,내가본영화가 뜻밖의여정이라니라 다섯군대전투네 ㅋㅋ 뜻밖의여정도봐야겟음.. 진짜최고 ...,NaN,NaN,NaN
5,8058669,정말 잠 온다 막장중에 막장,NaN,NaN,NaN
6,9074719,단순 지존파에 대한 얘기뿐만 아닌 대한민국 전체를 볼 수 있는 좋은 다큐,NaN,NaN,NaN
7,10034561,시간 때우기에 좋다...,NaN,NaN,NaN
8,8815532,이게 왜 유치하다고 할까 ? 어린이용 영화이고 또 내용도 상당히 잘된 작품이다. 씨...,NaN,NaN,NaN
9,10148006,God's Not Dead,NaN,NaN,NaN


## PA-04. Prepare Pilot Annotation

Manually annotate each review according to the multilingual sentiment 
annotation guideline.

For each review, record:

- **sentiment_label**: Sentiment assigned according to the guideline
- **confidence**: Confidence level of the annotation
- **note**: Reason for the decision or explanation of ambiguous cases

The pilot annotation is performed independently without referring to the original ground-truth labels.

In [12]:
pd.set_option("display.max_colwidth", None)

pilot_ko[[
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_id,review_body,sentiment_label,confidence,notes
0,10122209,미국이 진짜 선진국인 이유가 있구나 진짜 멋지드라...최고다,NaN,NaN,NaN
1,5849027,최고의 명작이라고 생각합니다.아직 못보신 분들은 꼭 감상하세요.,NaN,NaN,NaN
2,9643272,첨부터 끝까지 재밌음!!,NaN,NaN,NaN
3,1404443,먹먹한 상처와 조용히 다가오는 치유,NaN,NaN,NaN
4,10190504,내가본영화가 뜻밖의여정이라니라 다섯군대전투네 ㅋㅋ 뜻밖의여정도봐야겟음.. 진짜최고 반지의제왕시리즈는안봣지만..,NaN,NaN,NaN
5,8058669,정말 잠 온다 막장중에 막장,NaN,NaN,NaN
6,9074719,단순 지존파에 대한 얘기뿐만 아닌 대한민국 전체를 볼 수 있는 좋은 다큐,NaN,NaN,NaN
7,10034561,시간 때우기에 좋다...,NaN,NaN,NaN
8,8815532,이게 왜 유치하다고 할까 ? 어린이용 영화이고 또 내용도 상당히 잘된 작품이다. 씨리즈 뒤로 갈수록 산으로 갔지만 1-3 편은 명작이다.,NaN,NaN,NaN
9,10148006,God's Not Dead,NaN,NaN,NaN


In [13]:
pilot_ko.loc[0:4, [
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_id,review_body,sentiment_label,confidence,notes
0,10122209,미국이 진짜 선진국인 이유가 있구나 진짜 멋지드라...최고다,NaN,NaN,NaN
1,5849027,최고의 명작이라고 생각합니다.아직 못보신 분들은 꼭 감상하세요.,NaN,NaN,NaN
2,9643272,첨부터 끝까지 재밌음!!,NaN,NaN,NaN
3,1404443,먹먹한 상처와 조용히 다가오는 치유,NaN,NaN,NaN
4,10190504,내가본영화가 뜻밖의여정이라니라 다섯군대전투네 ㅋㅋ 뜻밖의여정도봐야겟음.. 진짜최고 반지의제왕시리즈는안봣지만..,NaN,NaN,NaN


In [15]:
pilot_ko["sentiment_label"] = pilot_ko["sentiment_label"].astype("object")
pilot_ko["confidence"] = pilot_ko["confidence"].astype("object")
pilot_ko["notes"] = pilot_ko["notes"].astype("object")

In [16]:
pilot_ko.loc[0, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[1, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[2, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[3, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]

pilot_ko.loc[3, "notes"] = "Implicit positive sentiment through healing theme"

In [17]:
pilot_ko.loc[0:3, [
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_body,sentiment_label,confidence,notes
0,미국이 진짜 선진국인 이유가 있구나 진짜 멋지드라...최고다,Positive,High,NaN
1,최고의 명작이라고 생각합니다.아직 못보신 분들은 꼭 감상하세요.,Positive,High,NaN
2,첨부터 끝까지 재밌음!!,Positive,High,NaN
3,먹먹한 상처와 조용히 다가오는 치유,Positive,Medium,Implicit positive sentiment through healing theme


In [18]:
print(pilot_ko.loc[4, "review_body"])

내가본영화가 뜻밖의여정이라니라 다섯군대전투네 ㅋㅋ 뜻밖의여정도봐야겟음.. 진짜최고 반지의제왕시리즈는안봣지만..


In [19]:
pilot_ko.loc[4, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [20]:
pilot_ko.loc[5:9, [
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_id,review_body,sentiment_label,confidence,notes
5,8058669,정말 잠 온다 막장중에 막장,NaN,NaN,NaN
6,9074719,단순 지존파에 대한 얘기뿐만 아닌 대한민국 전체를 볼 수 있는 좋은 다큐,NaN,NaN,NaN
7,10034561,시간 때우기에 좋다...,NaN,NaN,NaN
8,8815532,이게 왜 유치하다고 할까 ? 어린이용 영화이고 또 내용도 상당히 잘된 작품이다. 씨리즈 뒤로 갈수록 산으로 갔지만 1-3 편은 명작이다.,NaN,NaN,NaN
9,10148006,God's Not Dead,NaN,NaN,NaN


In [21]:
pilot_ko.loc[5, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[6, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_ko.loc[7, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[8, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_ko.loc[9, ["sentiment_label", "confidence"]] = ["Unclear", "High"]

pilot_ko.loc[9, "notes"] = "Title only; insufficient context to determine sentiment"

In [22]:
pilot_ko.loc[8, "notes"] = (
    "Contains mild negative wording, but positive evaluation dominates"
)

In [23]:
pilot_ko.loc[10:14, [
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_id,review_body,sentiment_label,confidence,notes
10,8083246,걍 레즈비언 영화네...,NaN,NaN,NaN
11,8769362,"하,,, 오락영화로서는 괜찮았던 1편이었는데, 2는 걍 이도저도 아닌 쓰레기",NaN,NaN,NaN
12,9874452,번외편 아니죠?ㅋㅋㅋ,NaN,NaN,NaN
13,9387278,경호원들은 밖에 무장병력이 많은거 알텐데 거기에다 딸랑 권총들고 닥돌하는거보면;,NaN,NaN,NaN
14,8005270,재미있었는데,NaN,NaN,NaN


In [24]:
pilot_ko.loc[10, ["sentiment_label", "confidence"]] = ["Unclear", "High"]
pilot_ko.loc[11, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[12, ["sentiment_label", "confidence"]] = ["Unclear", "Medium"]
pilot_ko.loc[13, ["sentiment_label", "confidence"]] = ["Unclear", "Low"]
pilot_ko.loc[14, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]

In [25]:
pilot_ko.loc[10, "notes"] = "Descriptive statement without explicit evaluation"

pilot_ko.loc[12, "notes"] = (
    "Insufficient context; laughter marker does not clearly indicate sentiment"
)

pilot_ko.loc[13, "notes"] = (
    "Criticism of a scene, but overall movie sentiment is unclear"
)

pilot_ko.loc[14, "notes"] = (
    "Positive expression, but the sentence appears incomplete"
)

In [26]:
for i in range(15, 20):
    print(f"[{i}] {pilot_ko.loc[i, 'review_body']}")
    print()

[15] 전쟁선포니 뭐니. 남북 위기고조가 되는 지금 이 시점에 디어평양을 보았습니다. 돌이킬수 없이 흘러가버린 한민족 열전이 얼마나 많은 상처를 우리에게 안겼는지. 사랑하는 이의 얼굴이 떠오릅니다.더이상의 이별은 없기를...

[16] 밑에 1점준 쓰레기들중 한국영화는 죄다 10점주는 M창들 많네 ㅋㅋ

[17] 와ㅡ다시보니 너무 재미있고 감동있고 그리고 생각해보니 한국영화에서 보지못한 신선한소재아이들과 같이봤는데 많은걸생각하고 너무 잼있게본영화.잘봤습니다.정재영님 연기 대박잘해ㅜㅡㅜ

[18] 다이애나를 평민으로 만든 영화

[19] 오유인들아 평점조작 쫌 그만 해달랑꼐?



In [27]:
pilot_ko.loc[15, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_ko.loc[16, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[17, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[18, ["sentiment_label", "confidence"]] = ["Neutral", "High"]
pilot_ko.loc[19, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]

In [28]:
pilot_ko.loc[15, "notes"] = (
    "Emotional appreciation without explicit movie evaluation"
)

pilot_ko.loc[16, "notes"] = (
    "Negative criticism directed at ratings/reviewers"
)

pilot_ko.loc[18, "notes"] = (
    "Descriptive statement without explicit sentiment"
)

pilot_ko.loc[19, "notes"] = (
    "Negative criticism of rating manipulation rather than the movie itself"
)

In [29]:
for i in range(20, 25):
    print(f"[{i}] {pilot_ko.loc[i, 'review_body']}")
    print()

[20] 귀여운 제니퍼 하는짓은 별로 안귀엽네.

[21] 마음에 듭니다.

[22] 정말 실타

[23] "* 많고많은 유령영화 ""넌 내가 사람으로 보이니 ~~~ """

[24] 문근영 서우 천정명 옥택연 모두 다 너무 좋았다 ㅠㅠ



In [30]:
pilot_ko.loc[20, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[21, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[22, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[23, ["sentiment_label", "confidence"]] = ["Unclear", "Low"]
pilot_ko.loc[24, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [31]:
pilot_ko.loc[23, "notes"] = (
    "Incomplete or context-poor text; sentiment cannot be reliably determined"
)

In [32]:
pilot_ko.loc[20, "notes"] = (
    "Negative evaluation dominates; not balanced enough for Mixed"
)

In [33]:
for i in range(25, 30):
    print(f"[{i}] {pilot_ko.loc[i, 'review_body']}")
    print()

[25] 오랜만에 다시봤다 언재 봐도 명작ㅎㅎ

[26] 이영화보고 빢쳐서 별점깍으러왓음ㅡㅡ

[27] 흥미로 점수 주는거 아니다 인정할건 인정하는 자세도 필요

[28] 4편까지 만들어서 망신 당하는 꼴은 안 봤으면.

[29] 원영의..ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ



In [34]:
pilot_ko.loc[25, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_ko.loc[26, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[27, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_ko.loc[28, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_ko.loc[29, ["sentiment_label", "confidence"]] = ["Unclear", "Low"]

In [35]:
pilot_ko.loc[27, "notes"] = (
    "Critical sentiment dominates, but interpretation is required"
)

pilot_ko.loc[29, "notes"] = (
    "Insufficient context; laughter alone does not determine sentiment"
)

## PA-06. Korean Pilot QA Check

Perform a quality check on the completed Korean pilot annotations.

The QA check verifies:

- No missing sentiment labels
- No missing confidence values
- Sentiment label distribution 
- Confidence distribution
- Review of ambiguous or low-confidence cases

In [36]:
print("Missing sentiment labels:",
      pilot_ko["sentiment_label"].isna().sum())

print("Missing confidence values:",
      pilot_ko["confidence"].isna().sum())

print("Missing notes:",
      pilot_ko["notes"].isna().sum())

Missing sentiment labels: 0
Missing confidence values: 0
Missing notes: 15


In [37]:
sentiment_distribution = (
    pilot_ko["sentiment_label"]
    .value_counts()
)

sentiment_distribution

sentiment_label
Positive    14
Negative     9
Unclear      6
Neutral      1
Name: count, dtype: int64

In [38]:
confidence_distribution = (
    pilot_ko["confidence"]
    .value_counts()
)

confidence_distribution

confidence
High      19
Medium     8
Low        3
Name: count, dtype: int64

In [39]:
qa_cases = pilot_ko[
    (pilot_ko["confidence"] == "Low") |
    (pilot_ko["sentiment_label"].isin(["Mixed", "Unclear"]))
]

qa_cases[[
    "review_id",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]]

,review_id,review_body,sentiment_label,confidence,notes
9,10148006,God's Not Dead,Unclear,High,Title only; insufficient context to determine sentiment
10,8083246,걍 레즈비언 영화네...,Unclear,High,Descriptive statement without explicit evaluation
12,9874452,번외편 아니죠?ㅋㅋㅋ,Unclear,Medium,Insufficient context; laughter marker does not clearly indicate sentiment
13,9387278,경호원들은 밖에 무장병력이 많은거 알텐데 거기에다 딸랑 권총들고 닥돌하는거보면;,Unclear,Low,"Criticism of a scene, but overall movie sentiment is unclear"
23,8605057,"""* 많고많은 유령영화 """"넌 내가 사람으로 보이니 ~~~ """"""",Unclear,Low,Incomplete or context-poor text; sentiment cannot be reliably determined
29,4644445,원영의..ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ,Unclear,Low,Insufficient context; laughter alone does not determine sentiment


## PA-07. Save Korean Pilot Annotation

Save the completed Korean pilot annotations for subsequent QA,
agreement analysis, and guideline revision.

In [40]:
# Define output path
output_path = Path(
    "../data/annotation/nsmc_ko_pilot_annotation_30.csv"
)

# Save pilot annotation
pilot_ko.to_csv(
    output_path,
    index=False,
    encoding="utf-8-sig"
)

print("Saved to:", output_path)
print("File exists:", output_path.exists())

Saved to: ..\data\annotation\nsmc_ko_pilot_annotation_30.csv
File exists: True


## PA-08. Korean Pilot Finding & Guideline Issues

The Korean pilot annotation identified several ambiguous cases that require 
clarification in the next version of the annotation guideline.

### 1. Neutral vs. Unclear

The distinction between `Neutral` and `Unclear` requires clearer rules.

- `Neutral`: The meaning of the text is understandable, but no clear positive or negative evaluation is expressed.
- `Unclear`: The sentiment cannot reliably determined because of 
insufficient context, incomplete text, or ambiguous meaning.

**Pilot examples:**
-`"다이애나를 평민으로 만든 영화"`→ `Unclear / High`
 - Title only; insufficient information to determine sentiment.

 **Guideline revision candidate:**
 Add an explicit decision rule distinguishing descriptive but interpretable 
 reviews from context-insufficient reviews.

 ---

 ### 2. Confidence for Unclear Labels

 The pilot revealed ambiguity in assigning confidence to `Unclear` labels.

 An annotation may be confidently classified as `Unclear` when the annotator
 is certain that the available text does not provide enough evidence for
 sentiment classification.

 Examples:

 - Title-only review → `Unclear / High`
 - Ambiguous laughter or incomplete expression → `Unclear / Low`

 **Guideline revision candidate:**
 Clarify that confidence represents confidence in the annotation decision, 
 not the strength or clarify of the sentiment itself.

 ### 3. Sentiment Target

 Some reviews express sentiment toward ratings, reviewers, characters,
 or individual scenes rather than toward the movie itself.

 **Pilot examples:**

 - Criticism of rating manipulation
 - Criticism of other reviewers
 - Criticism of a specific movie scene

 **Guideline revision candidate:**
 Define whether sentiment should reflect: 

 1. the overall emotional polarity expressed in the review, or
 2. sentiment specifically directed toward the reviewed product/movie.

 A consistent target rule should be established before main annotation.

 ---

 ### 4. Incomplete and Context-Poor Reviews 

 Short or incomplete reviews frequently require additional interpretation.

 Examples include:

 - incomplete sentences
 - isolated movie titles 
 - quotations without sufficient context
 - short expression followed only by `ㅋㅋ` or `ㅎㅎ`

 **Guideline revision candidate:**
 Add more examples for distinguishing `Neutral` from `Unclear` in
 short and incomplete reviews.

 ### 5. Korean Internet Expressions

 Korean laughter markers such as `ㅋㅋ` and `ㅎㅎ` do not independently indicate positive sentiment.

 Their interpretation depends on surrounding context and may represent: 

 - amusement 
 - sarcasm
 - ridicule
 - embarrassment
 - no clear sentiment

 **Guideline revision candidate:**
 Expand the Korean-specific section with examples of laughter markers 
 used in positive, negative, sarcastic, and unclear contexts.

---

### Korean Pilot Summary

- Pilot samples: **30**
- Positive: **14 (46.7%)**
- Negative: **9 (30.0%)**
- Neutral: **1 (3.3%)**
- Unclear: **6 (20.0%)**
- Mixed: **0 (0.0%)**

Confidence distribution:

- High: **19 (63.3%)**
- Medium: **8 (26.7%)**
- Low: **3 (10.0%)**

The Korean pilot suggests that the main areas requiring clarification are
`Neutral vs. Unclear`, confidence assignment for `Unclear`, sentiment target,
and language-specific interpretation of short or ambiguous expressions.


## PA-09. Load English Annotation Data

Load the prepared English Amazon review dataset for pilot annotation.

In [41]:
# Define English annotation file path
en_path = Path(
    "../data/annotation/amazon_reviews_en_annotation_300.csv"
)

# Load dataset
df_en = pd.read_csv(en_path)

# Preview data
df_en.head()

,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,en_0331970,Good cord,Quick delivery works as it should. Thank you,en,other,NaN,NaN,NaN
1,en_0897981,Perfect lights.,"We have these on the driveway, garage and on the fence in the back yard. No need to leave the back light on anymore when we go away. Surprises anyone who is lurking around and not expecting it too :D They are bright and helps when I am unloading groceries from the car at night. Planning to get at least 1 more for the back yard, maybe 2.",en,home_improvement,NaN,NaN,NaN
2,en_0057491,Great battery life.,"I have many ear buds, the Peats sound great. The battery lasts all day. I have not tried Alex yet. They are not uncomfortable after using all day. The only improvement I would recommend is a better clasp to clip on to your shirt.",en,wireless,NaN,NaN,NaN
3,en_0711669,no. just no.,"The other reviews warned about the manual. they were right, it basically just tells you that it's a camera. There are 2 buttons, one on each side with the letter M. neither of which brings you to menu. So it's difficult to set the time or use any of it's features. I will be returning mine.",en,other,NaN,NaN,NaN
4,en_0873342,good blades,"Watch the online video on how to replace blades. Once I did, it was easy and fast.",en,automotive,NaN,NaN,NaN


In [42]:
print("File exists:", en_path.exists())
print("Dataset shape:", df_en.shape)
print("Columns:", df_en.columns.tolist())

df_en.info()

File exists: True
Dataset shape: (300, 8)
Columns: ['review_id', 'review_title', 'review_body', 'language', 'product_category', 'sentiment_label', 'confidence', 'notes']
<class 'pandas.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   review_id         300 non-null    str    
 1   review_title      300 non-null    str    
 2   review_body       300 non-null    str    
 3   language          300 non-null    str    
 4   product_category  300 non-null    str    
 5   sentiment_label   0 non-null      float64
 6   confidence        0 non-null      float64
 7   notes             0 non-null      float64
dtypes: float64(3), str(5)
memory usage: 83.9 KB


In [43]:
# Create a reproducible English pilot sample
pilot_en = df_en.sample(
    n=30,
    random_state=42
).copy()

# Reset index
pilot_en = pilot_en.reset_index(drop=True)

In [44]:
pilot_en["sentiment_label"] = (
    pilot_en["sentiment_label"].astype("object")
)

pilot_en["confidence"] = (
    pilot_en["confidence"].astype("object")
)

pilot_en["notes"] = (
    pilot_en["notes"].astype("object")
)

In [45]:
print("English pilot sample shape:", pilot_en.shape)

pilot_en[[
    "review_id",
    "review_title",
    "review_body",
    "sentiment_label",
    "confidence",
    "notes"
]].head()

English pilot sample shape: (30, 8)


,review_id,review_title,review_body,sentiment_label,confidence,notes
0,en_0878143,Nice try on my part,Disappointed in the amount of cash it can hold. Difficult to get cards in and out,NaN,NaN,NaN
1,en_0165695,NOT like described,"The colors are not very ""assorted"" they are very similar in color. Not like the picture shown at all.",NaN,NaN,NaN
2,en_0817116,Runs small,Very cute but made small. I ordered a size larger than I normally wear and it was tight.,NaN,NaN,NaN
3,en_0099953,Dependability and price,Provides the hold I need on a daily bases,NaN,NaN,NaN
4,en_0720821,Works great. I plan on buying more,Works great. I plan on buying more.,NaN,NaN,NaN


In [46]:
pilot_en.loc[0, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[1, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[2, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_en.loc[3, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_en.loc[4, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [47]:
pilot_en.loc[2, "notes"] = (
    "Positive comment on appearance, but sizing issue dominates overall sentiment"
)

pilot_en.loc[3, "notes"] = (
    "Practical satisfaction expressed without strong emotional wording"
)

In [48]:
for i in range(5, 10):
    print(f"[{i}]")
    print("Title:", pilot_en.loc[i, "review_title"])
    print("Body :", pilot_en.loc[i, "review_body"])
    print()

[5]
Title: Good but won't solve all your problems
Body : I bought this product because I thought it would help me not get a flat tire. There are a lot of thorns in my area. Unfortunately it does not help against nails. For the most part it works but it's still not strong enough. I am still looking for the perfect tire liner.

[6]
Title: Does not work!
Body : This shampoo did not remove orange from my hair, in fact, it brought back orange! It stripped the dark ash blonde dye from my hair and left it back in its orange state. My hair felt very soft in the morning, that’s why I’m giving it 2 stars. But it did not tone the orange out at all, it made my hair even more orange. I will use it a few more times and if it starts working I will update my review.

[7]
Title: Fabric is stiff
Body : One of the prints wasn’t Christmas and the feel of the fabric is not soft.

[8]
Title: The sizing factor is incorrect. If you need a Small, order a Medium.
Body : The sizing factor is incorrect. If you ne

In [49]:
pilot_en.loc[5, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[6, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[7, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[8, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_en.loc[9, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [50]:
pilot_en.loc[5, "notes"] = (
    "Some positive functionality, but product limitations dominate"
)

pilot_en.loc[6, "notes"] = (
    "Minor positive effect, but failure of the main product function dominates"
)

pilot_en.loc[8, "notes"] = (
    "Negative sizing issue expressed partly as practical advice"
)

In [51]:
for i in range(10, 15):
    print(f"[{i}]")
    print("Title:", pilot_en.loc[i, "review_title"])
    print("Body :", pilot_en.loc[i, "review_body"])
    print()

[10]
Title: One Star
Body : I ordered a 2 pack and only received one pack!

[11]
Title: Broken case and loose disc..
Body : Well i love this movie but the inside of the case was broke and the disc was scratched up from being loose in the case. Really disappointed

[12]
Title: good quality
Body : These are very good quality, and the arch support is good for my old feet. However, the heel is a bit loose for my narrow feet. I would not be able to wear the slippers without socks.

[13]
Title: Good charger on the go.
Body : Charger has worked fine for my needs. The fan is very loud as is the end of cycle beep but noise aside, it's great.

[14]
Title: Good video monitor camera
Body : Camera works great in our baby's room. Have no complaints other than the temp gauge seems to fluctuate quite a bit.



In [52]:
pilot_en.loc[10, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[11, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[12, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_en.loc[13, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_en.loc[14, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]

In [53]:
pilot_en.loc[11, "notes"] = (
    "Positive opinion of the movie, but damaged product condition dominates"
)

pilot_en.loc[12, "notes"] = (
    "Minor fit issue, but overall product quality is evaluated positively"
)

pilot_en.loc[13, "notes"] = (
    "Noise complaints are outweighed by overall positive performance"
)

pilot_en.loc[14, "notes"] = (
    "Minor temperature gauge issue, but overall evaluation remains positive"
)

In [54]:
for i in range(15, 20):
    print(f"[{i}]")
    print("Title:", pilot_en.loc[i, "review_title"])
    print("Body :", pilot_en.loc[i, "review_body"])
    print()

[15]
Title: Slick Willy sez:
Body : First off, been buying this stuff for years-the wife and I go through it like some people buy milk, if ya know what I mean. Anyway, the bottles leak worse than my wife when she sees me naked. So, bad design. I'm thinking of wrapping some teflon tape on it. On the threads of the bottle, that is. The lube itself is top-notch and lasts for hours but it gets expensive. I spent more on this last year than on my kid's college fund. But in all fairness, it's a better investment. It also stays cool and maintains sufficient lubricity even in the high heat that is generated in extreme friction situations. So much so that, while I have not yet done it, I have pondered replacing the Mobil 1 full-synthetic oil in my truck with this stuff. So if you like to have the excitement, and who doesn't, get yourself some of this.

[16]
Title: Not the best tasting in my opinion
Body : Just not crazy about the flavor.

[17]
Title: Don’t Waste Your $!!!
Body : Ring is beautif

In [55]:
pilot_en.loc[15, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_en.loc[16, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[17, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[18, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_en.loc[19, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]

In [56]:
pilot_en.loc[15, "notes"] = (
    "Bottle design and price issues are outweighed by strong product praise"
)

pilot_en.loc[18, "notes"] = (
    "Short-term odor removal works, but poor scent longevity dominates"
)

pilot_en.loc[19, "notes"] = (
    "Some positive features, but multiple fit and usability complaints dominate"
)

In [57]:
for i in range(20, 25):
    print(f"[{i}]")
    print("Title:", pilot_en.loc[i, "review_title"])
    print("Body :", pilot_en.loc[i, "review_body"])
    print()

[20]
Title: One Star
Body : I have this stool not even one year, the hydraulic bar is not functioning

[21]
Title: Great product..great price
Body : Arrived before scheduled. Did as directed on instructions. Works beautifully and is a very soothing ambiance

[22]
Title: Did not get instruction manual
Body : I expected cheap, but at the very minimum expected instructions. None were included. This ended up being so frustrating!

[23]
Title: The problem is good
Body : The product is good but my order was short of a soap

[24]
Title: Too small for AR15
Body : Bought this for cleaning an AR15, but it’s too small, had to return.



In [58]:
pilot_en.loc[20, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[21, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_en.loc[22, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[23, ["sentiment_label", "confidence"]] = ["Mixed", "Medium"]
pilot_en.loc[24, ["sentiment_label", "confidence"]] = ["Negative", "High"]

In [59]:
pilot_en.loc[23, "notes"] = (
    "Positive product evaluation combined with negative order fulfillment experience"
)

In [60]:
for i in range(25, 30):
    print(f"[{i}]")
    print("Title:", pilot_en.loc[i, "review_title"])
    print("Body :", pilot_en.loc[i, "review_body"])
    print()

[25]
Title: No cords to link them together
Body : Just arrived. Lights seem alright and quite bright. 3 stars because the cords for hooking the lights up in a series are not included. The way it is written up I was definitely expecting the cords to come with the lights. So now I need to find the right cords.

[26]
Title: Poor handling of package
Body : I received the box in reasonable time. The box came severely damaged. It looked like it was used as a football. I didn’t wan to take a chance on receiving a damaged product so I opted to return it in the box it came in. I received the return sticker from ups the next day. The instructions as to what to do from there got sloppy. I had to jump through hoops to get the item back to amazon. Home pick up was supposed to be available but when I logged in it told me to take it to a local ups drop off. Went there only to find when I entered the info it was rejected and they directed me to a manned drop off. Very inconvenient. My last few amazon 

In [61]:
pilot_en.loc[25, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[26, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[27, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[28, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_en.loc[29, ["sentiment_label", "confidence"]] = ["Negative", "High"]

In [62]:
pilot_en.loc[25, "notes"] = (
    "Positive comment on brightness, but missing expected cords dominates"
)

pilot_en.loc[29, "notes"] = (
    "Minor positive comment on appearance, but severe usability and assembly failures dominate"
)

## PA-11 English Pilot QA Check

Perform quality assurance checks on the completed English pilot annotations.

QA checks include:
- Missing sentiment labels
- Missing confidence values
- Sentiment label distribution
- Confidence distribution
- Review of Mixed, Unclear, and Low-confidence cases

In [63]:
# Check missing annotation fields
print("Missing sentiment labels:",
      pilot_en["sentiment_label"].isna().sum())

print("Missing confidence values:",
      pilot_en["confidence"].isna().sum())

print("Missing notes:",
      pilot_en["notes"].isna().sum())

Missing sentiment labels: 0
Missing confidence values: 0
Missing notes: 15


In [64]:
# Sentiment label distribution
sentiment_distribution = (
    pilot_en["sentiment_label"]
    .value_counts(dropna=False)
)

print("Sentiment Label Distribution")
print(sentiment_distribution)

print("\nSentiment Label Percentage")
print(
    pilot_en["sentiment_label"]
    .value_counts(normalize=True, dropna=False)
    .mul(100)
    .round(1)
)

Sentiment Label Distribution
sentiment_label
Negative    21
Positive     8
Mixed        1
Name: count, dtype: int64

Sentiment Label Percentage
sentiment_label
Negative    70.0
Positive    26.7
Mixed        3.3
Name: proportion, dtype: float64


In [65]:
# Confidence distribution
confidence_distribution = (
    pilot_en["confidence"]
    .value_counts(dropna=False)
)

print("Confidence Distribution")
print(confidence_distribution)

print("\nConfidence Percentage")
print(
    pilot_en["confidence"]
    .value_counts(normalize=True, dropna=False)
    .mul(100)
    .round(1)
)

Confidence Distribution
confidence
High      21
Medium     9
Name: count, dtype: int64

Confidence Percentage
confidence
High      70.0
Medium    30.0
Name: proportion, dtype: float64


In [66]:
# Extract cases requiring additional QA review
qa_cases_en = pilot_en[
    (pilot_en["confidence"] == "Low") |
    (pilot_en["sentiment_label"].isin(["Mixed", "Unclear"]))
].copy()

print("Number of QA cases:", len(qa_cases_en))

qa_cases_en[
    [
        "review_id",
        "review_title",
        "review_body",
        "sentiment_label",
        "confidence",
        "notes"
    ]
]

Number of QA cases: 1


,review_id,review_title,review_body,sentiment_label,confidence,notes
23,en_0561531,The problem is good,The product is good but my order was short of a soap,Mixed,Medium,Positive product evaluation combined with negative order fulfillment experience


### English Pilot QA Summary

- Total pilot samples: **30**
- Missing sentiment labels: **0**
- Missing confidence values: **0**
- Negative: **21 (70.0%)**
- Positive: **8 (26.7%)**
- Mixed: **1 (3.3%)**
- Neutral: **0**
- Unclear: **0**
- High confidence: **21 (70.0%)**
- Medium confidence: **9 (30.0%)**
- Low confidence: **0**
- Additional QA cases: **1**

#### QA Review

One `Mixed / Medium` case was identified for additional review.

The review contains a clearly positive product evaluation but a negative
order-fulfillment experience. Since both sentiments are meaningful and
neither clearly eliminates the other, the `Mixed / Medium` annotation
was retained.

No annotation changes were required after QA.

## PA-12 Save English Pilot Annotation

Save the completed and QA-reviewed English pilot annotations as a
separate CSV file while preserving the original 300-sample annotation
dataset for the main annotation stage.

In [67]:
# Save English pilot annotation

output_path_en = Path(
    "../data/annotation/amazon_reviews_en_pilot_annotation_30.csv"
)

pilot_en.to_csv(
    output_path_en,
    index=False,
    encoding="utf-8-sig"
)

print("Saved to:", output_path_en)
print("File exists:", output_path_en.exists())

Saved to: ..\data\annotation\amazon_reviews_en_pilot_annotation_30.csv
File exists: True


In [68]:
# Verify saved pilot annotation

saved_pilot_en = pd.read_csv(output_path_en)

print("Saved dataset shape:", saved_pilot_en.shape)
print(
    "Missing sentiment labels:",
    saved_pilot_en["sentiment_label"].isna().sum()
)
print(
    "Missing confidence values:",
    saved_pilot_en["confidence"].isna().sum()
)

saved_pilot_en.head()

Saved dataset shape: (30, 8)
Missing sentiment labels: 0
Missing confidence values: 0


,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,en_0878143,Nice try on my part,Disappointed in the amount of cash it can hold. Difficult to get cards in and out,en,apparel,Negative,High,NaN
1,en_0165695,NOT like described,"The colors are not very ""assorted"" they are very similar in color. Not like the picture shown at all.",en,office_product,Negative,High,NaN
2,en_0817116,Runs small,Very cute but made small. I ordered a size larger than I normally wear and it was tight.,en,apparel,Negative,Medium,"Positive comment on appearance, but sizing issue dominates overall sentiment"
3,en_0099953,Dependability and price,Provides the hold I need on a daily bases,en,other,Positive,Medium,Practical satisfaction expressed without strong emotional wording
4,en_0720821,Works great. I plan on buying more,Works great. I plan on buying more.,en,home_improvement,Positive,High,NaN


## PA-13 English Pilot Findings & Guideline Issues

The English pilot annotation identified several cases that require
additional clarification in the next version of the annotation guideline.

### 1. Dominant Sentiment in Reviews with Both Positive and Negative Comments

Several reviews contained both positive and negative expressions.
The number of positive or negative phrases alone was not sufficient
to determine the overall sentiment.

The annotation decision should consider:

- the main purpose of the review,
- the importance of the product feature being evaluated,
- the review title and conclusion,
- whether the positive or negative issue dominates the overall experience.

**Example**

> "Ring is beautiful but not sterling silver... Very disappointed."

Although the review contains a positive comment about appearance,
the product-quality problem and explicit disappointment dominate.

**Decision:** `Negative / High`

**Guideline issue:**  
Add a clearer dominance rule explaining that sentiment should be
determined by the overall evaluation rather than by counting
positive and negative expressions.

---

### 2. Core Product Function vs. Minor Positive or Negative Features

Some reviews included minor positive comments while reporting failure
of the product's main function.

For example, a product may have good appearance or secondary benefits
while failing to perform its primary purpose.

**Decision principle:**  
Failure of a core product function may outweigh minor positive
features when determining overall sentiment.

**Guideline issue:**  
Clarify how annotators should weigh core functionality against
secondary product attributes.

---

### 3. Product Sentiment vs. Purchase or Fulfillment Experience

Some reviews evaluated the product positively but described a negative
purchase, packaging, delivery, or order-fulfillment experience.

**Example**

> "The product is good but my order was short of a soap."

The product evaluation is positive, while the fulfillment experience
is negative.

**Decision:** `Mixed / Medium`

**Guideline issue:**  
Define whether sentiment annotation should cover:

- the product only, or
- the complete customer experience including delivery, packaging,
  fulfillment, and customer service.

A consistent sentiment target should be established for future
annotation.

---

### 4. Mixed vs. Dominant Sentiment

The English pilot produced one `Mixed` case.

`Mixed` should be used when meaningful positive and negative
evaluations are both present and neither clearly dominates.

However, the presence of words such as "but", "however", or "although"
does not automatically make a review Mixed.

**Example**

> "Works great... However, the temperature gauge fluctuates."

The minor negative issue does not outweigh the overall positive
evaluation.

**Decision:** `Positive / Medium`

**Guideline issue:**  
Add more contrastive examples showing the difference between:

- `Mixed`
- `Positive with minor negative comments`
- `Negative with minor positive comments`

---

### 5. Confidence Reflects Annotation Certainty

The English pilot contained:

- High confidence: 21 (70.0%)
- Medium confidence: 9 (30.0%)
- Low confidence: 0

The absence of Low-confidence cases does not indicate a QA problem.
Confidence should represent the annotator's certainty about the
assigned label rather than sentiment intensity.

**Guideline issue:**  
Explicitly distinguish:

> Confidence in the annotation decision

from

> Strength or intensity of the sentiment.

---

### 6. English Pilot Summary

The English pilot consisted of 30 reviews.

Final sentiment distribution:

- Negative: 21 (70.0%)
- Positive: 8 (26.7%)
- Mixed: 1 (3.3%)
- Neutral: 0
- Unclear: 0

QA identified one additional review case (`Mixed / Medium`), and the
annotation was retained after review.

No missing sentiment labels or confidence values were found.

### Proposed Guideline v0.2 Updates

Based on the English pilot, the next guideline revision should consider:

1. Expanding the dominant-sentiment decision rule.
2. Clarifying how core product functionality should be weighted.
3. Defining the sentiment target: product vs. complete customer experience.
4. Adding clearer Mixed vs. dominant-sentiment examples.
5. Clarifying that confidence represents annotation certainty rather
   than sentiment intensity.

These findings will be combined with the Korean and Chinese pilot
results before updating the guideline from v0.1 to v0.2.

## PA-14 Load Chinese Annotation Data

Load the Chinese annotation dataset and prepare it for pilot annotation.

In [69]:
# Load Chinese annotation dataset

zh_path = Path(
    "../data/annotation/amazon_reviews_zh_annotation_300.csv"
)

df_zh = pd.read_csv(zh_path)

print("File exists:", zh_path.exists())
print("Dataset shape:", df_zh.shape)
print("Columns:", df_zh.columns.tolist())

df_zh.info()

File exists: True
Dataset shape: (300, 8)
Columns: ['review_id', 'review_title', 'review_body', 'language', 'product_category', 'sentiment_label', 'confidence', 'notes']
<class 'pandas.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   review_id         300 non-null    str    
 1   review_title      300 non-null    str    
 2   review_body       300 non-null    str    
 3   language          300 non-null    str    
 4   product_category  300 non-null    str    
 5   sentiment_label   0 non-null      float64
 6   confidence        0 non-null      float64
 7   notes             0 non-null      float64
dtypes: float64(3), str(5)
memory usage: 75.9 KB


In [70]:
# Create Chinese pilot sample

pilot_zh = df_zh.sample(
    n=30,
    random_state=42
).copy()

pilot_zh = pilot_zh.reset_index(drop=True)

print("Chinese pilot shape:", pilot_zh.shape)

pilot_zh.head()

Chinese pilot shape: (30, 8)


,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,zh_0842626,质量实在一般,拿到手非常喜欢，轻便，简单，耐看。但是质量很p，五一假期穿着跟朋友玩了会儿篮球斗牛，结果左脚小白鞋直接坏掉：后跟脱胶，前面鞋面蹭破。此外，感觉时间一长大拇指会把鞋面顶破，质量实在不敢恭维。我穿三叶草打篮球搞运动什么的，完全不会出现一丢丢问题。,zh,shoes,NaN,NaN,NaN
1,zh_0531516,服务态度超好！,服务态度超好，衣服质量也是杠杠的，谢谢，祝生意越做越好！,zh,apparel,NaN,NaN,NaN
2,zh_0474460,刑法学,书还不错 纸张也还好 发货到货都快。,zh,book,NaN,NaN,NaN
3,zh_0915261,书可以，快递太差,周四下单时显示有货，周六送到，还特地请了假等书，结果周六凌晨发邮件给我说没货，下周一送到，然后没通知的情况下快递又周日送到，我不在，还被快递大骂一顿，说不在怎么不选工作日送货,zh,book,NaN,NaN,NaN
4,zh_0581507,好书，值得推荐！,昨天买的，今天就送到了，大致流浪了一下，里面的问题都是在教育孩子的过程中遇到的典型问题，好书，值得推荐！,zh,book,NaN,NaN,NaN


In [71]:
# Prepare annotation columns for manual labeling

pilot_zh["sentiment_label"] = (
    pilot_zh["sentiment_label"].astype("object")
)

pilot_zh["confidence"] = (
    pilot_zh["confidence"].astype("object")
)

pilot_zh["notes"] = (
    pilot_zh["notes"].astype("object")
)

In [72]:
for i in range(0, 5):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[0]
Title: 质量实在一般
Body : 拿到手非常喜欢，轻便，简单，耐看。但是质量很p，五一假期穿着跟朋友玩了会儿篮球斗牛，结果左脚小白鞋直接坏掉：后跟脱胶，前面鞋面蹭破。此外，感觉时间一长大拇指会把鞋面顶破，质量实在不敢恭维。我穿三叶草打篮球搞运动什么的，完全不会出现一丢丢问题。

[1]
Title: 服务态度超好！
Body : 服务态度超好，衣服质量也是杠杠的，谢谢，祝生意越做越好！

[2]
Title: 刑法学
Body : 书还不错 纸张也还好 发货到货都快。

[3]
Title: 书可以，快递太差
Body : 周四下单时显示有货，周六送到，还特地请了假等书，结果周六凌晨发邮件给我说没货，下周一送到，然后没通知的情况下快递又周日送到，我不在，还被快递大骂一顿，说不在怎么不选工作日送货

[4]
Title: 好书，值得推荐！
Body : 昨天买的，今天就送到了，大致流浪了一下，里面的问题都是在教育孩子的过程中遇到的典型问题，好书，值得推荐！



In [73]:
pilot_zh.loc[0, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[1, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[2, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[3, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[4, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [74]:
pilot_zh.loc[0, "notes"] = (
    "Positive comments on appearance and comfort, but serious quality problems dominate"
)

pilot_zh.loc[2, "notes"] = (
    "Mild positive wording, but the overall positive sentiment is unambiguous"
)

pilot_zh.loc[3, "notes"] = (
    "Product itself is acceptable, but negative delivery and service experience dominates"
)

In [75]:
for i in range(5, 10):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[5]
Title: 质量差
Body : 矬是弯的，每次打开合上都会磨主刀。镊子两头不一样长，夹个小东西那个费劲啊。

[6]
Title: 价格很合理
Body : 洗的時候会褪色，但过了几遍水就不会了，布料很好，不会厚，是我想要的

[7]
Title: 书质量差
Body : 数的质量好差，直接开裂了，内容还没有看。

[8]
Title: 还不错
Body : 故事还不错吧，但是标题洗澡和简介，真是感觉没有什么关系。

[9]
Title: 超值，喜欢
Body : 一直都使用这两个牌子，赶上双十一优惠，超值。



In [76]:
pilot_zh.loc[5, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[6, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[7, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[8, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_zh.loc[9, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [77]:
pilot_zh.loc[6, "notes"] = (
    "Minor color-fading issue, but positive fabric evaluation and overall satisfaction dominate"
)

pilot_zh.loc[8, "notes"] = (
    "Positive evaluation of the story, with a secondary criticism of the title and description"
)

In [78]:
for i in range(10, 15):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[10]
Title: 书皮破
Body : 一打开包装就看见了破烂的包装书皮，给人一种二手货的感觉。不开心。

[11]
Title: 怪癖心理学
Body : 内容写得通俗易懂还有小案例，但是觉得有点笼统，没有达到自己期望

[12]
Title: 发过来的新鞋子有磨损，无法接受
Body : 鞋是好鞋，请问亚马逊海外直邮是怎么包装的，直接裸盒套个塑料袋就发给我了？鞋盒都破了，鞋子也刮花了，你们就这么服务客户？

[13]
Title: kindle版排版错误太多了。
Body : 书是很棒了，毕竟等了那么久的电子版。 但是中间的低级错误太多了。错别字和本编排错误。 那些文学大家的名字错误的最多。 希望可以改善

[14]
Title: 老爷的漫画真心精品啊
Body : 从包装封面到书的内容都很棒！这本收藏必入啊！！



In [79]:
pilot_zh.loc[10, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[11, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_zh.loc[12, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[13, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_zh.loc[14, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [80]:
pilot_zh.loc[11, "notes"] = (
    "Positive comments on readability, but failure to meet expectations dominates"
)

pilot_zh.loc[12, "notes"] = (
    "Product itself is positively evaluated, but packaging damage and service complaints dominate"
)

pilot_zh.loc[13, "notes"] = (
    "Book content is positively evaluated, but numerous Kindle formatting and text errors dominate"
)

In [81]:
for i in range(15, 20):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[15]
Title: 很实用
Body : 锅锅不错，比我想象中要大，材质很好，卖家还送了小赠品，很实用，超级满意的，无意间看到，就来评价了，的确不错的。

[16]
Title: 亚马逊怎么了？
Body : 8月29下单到现在还没有发货，亚马逊是怎么了？

[17]
Title: 质量好
Body : 质量好又便宜，五功能有点鸡肋，只用一种功能，其它的我感觉没什么用，有一个功能不出水我以为坏的，问了品牌商客服原来是滴水功能，就是洗头发的时候节水。我用的出水面积最大的那个功能，水打在身上很温柔，加上家里水压不足，一开始有点不习惯，用久了觉得好舒服。一个好的花洒能让你享受洗澡， 哈哈。

[18]
Title: 没有地线
Body : 作为松下品牌，灯具地线也没有，打电话给客服，没有任何的回复，失望，大家不要买

[19]
Title: 间歇性掉线。 信号也不强，不如家里旧的TPLINK。
Body : 间歇性掉线。 信号也不强，不如家里旧的TPLINK。



In [82]:
pilot_zh.loc[15, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[16, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[17, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[18, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[19, ["sentiment_label", "confidence"]] = ["Negative", "High"]

In [83]:
pilot_zh.loc[17, "notes"] = (
    "Several minor functional concerns are present, "
    "but overall product satisfaction clearly dominates"
)

In [84]:
pilot_zh.loc[16, "notes"] = (
    "Negative sentiment is clear from the shipping delay "
    "and rhetorical question despite no explicit negative adjective"
)

In [85]:
for i in range(20, 25):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[20]
Title: 以为可以媲美东野圭吾
Body : 以为可以媲美东野圭吾，不过是我的一厢情愿。故事很长，到中部已经扣不住人心，再看下去纯属想知道结局，没有倒吸一口冷气的那种震撼。

[21]
Title: 价格质量很好
Body : 满100减100优惠后79，应该接近成本价了吧，亚马逊真给力，希望多点这样的活动！

[22]
Title: 差评
Body : 杯子做的很轻薄，至于这个属于质量好还是差就不说了，关键是醒酒瓶上有一个非常明显的瑕疵，这个在市场上是卖不出去的那种老板怎么就给我发过来了呢，是在实体店里卖不了了所以赶紧在网上卖出去吧，太缺乏诚信。。退货吧太麻烦，算了，以后再不会光顾了。

[23]
Title: 购物体验好差
Body : 买了东西两天后（完完整整的49个小时）才发货，完全没有渠道问一下是什么情况。收到后拆封发现里面连个塑料袋都不套，怕了，真差劲，等安装吧，安装的时候看看货物情况

[24]
Title: 非常舒服的鞋
Body : 这双鞋最大的特点就是软，鞋底软，鞋垫软，皮面也软，整体也很轻，走路很舒服。



In [86]:
pilot_zh.loc[20, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[21, ["sentiment_label", "confidence"]] = ["Positive", "High"]
pilot_zh.loc[22, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[23, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[24, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [87]:
pilot_zh.loc[20, "notes"] = (
    "Negative sentiment is expressed through unmet expectations "
    "and loss of interest without strong explicit negative wording"
)

In [88]:
pilot_zh.loc[23, "notes"] = (
    "Negative sentiment is driven by shipping delay, "
    "lack of communication, and poor packaging experience"
)

In [89]:
for i in range(25, 30):
    print(f"[{i}]")
    print("Title:", pilot_zh.loc[i, "review_title"])
    print("Body :", pilot_zh.loc[i, "review_body"])
    print()

[25]
Title: 希望常有货
Body : 很好 就是总没货 这次比较慢 二十多天才到吧 之后下的英亚的都到了这个都还没到= =

[26]
Title: 发错货了！！！里面没有完形填空这本书！！
Body : 打开包装膜，没有完形填空，请解决一下！！！！！

[27]
Title: 很好
Body : 很好。以前从不去评价，不知道浪费了多少积分，现在知道积分可以换钱，就要好好评价了，后来我就把这段话复制走了，既能赚积分，还省事，走到哪复制到哪，直接发出就可以了，推荐给大家！

[28]
Title: 鉴定完毕-这家出版不值得买
Body : 读到一半左右往后，印刷看上去很费劲，像是缺墨一样，伤眼睛。

[29]
Title: 还是比较满意的一次网购
Body : 一.快递速度很快，比预计早了两天到，一星 二.商品是正品，防伪，保修，发票等都有 ，二星 三.包装的很严实，没有破损 ，三星 四，和商品描述的差不多，读写和写入速度都在可接受范围 ，四星 五，总的来说没有毛病，nice 五星



In [90]:
pilot_zh.loc[25, ["sentiment_label", "confidence"]] = ["Negative", "Medium"]
pilot_zh.loc[26, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[27, ["sentiment_label", "confidence"]] = ["Positive", "Medium"]
pilot_zh.loc[28, ["sentiment_label", "confidence"]] = ["Negative", "High"]
pilot_zh.loc[29, ["sentiment_label", "confidence"]] = ["Positive", "High"]

In [91]:
pilot_zh.loc[25, "notes"] = (
    "Positive product comment, but stock availability "
    "and long delivery delay dominate the review"
)

pilot_zh.loc[27, "notes"] = (
    "Positive wording is present, but most of the review "
    "is generic text copied for reward points"
)

In [92]:
pilot_zh.loc[28, "notes"] = (
    "Poor printing quality causes readability problems "
    "and leads to an explicit negative recommendation"
)

## PA-17 Chinese Pilot QA Check

Perform quality assurance checks on the completed Chinese pilot annotations.

QA checks include:
- Missing sentiment labels
- Missing confidence values
- Sentiment label distribution
- Confidence distribution
- Review of Mixed, Unclear, and Low-confidence cases

In [93]:
# Check missing annotation fields

print(
    "Missing sentiment labels:",
    pilot_zh["sentiment_label"].isna().sum()
)

print(
    "Missing confidence values:",
    pilot_zh["confidence"].isna().sum()
)

print(
    "Missing notes:",
    pilot_zh["notes"].isna().sum()
)

Missing sentiment labels: 0
Missing confidence values: 0
Missing notes: 15


In [94]:
# Sentiment label distribution

sentiment_distribution_zh = (
    pilot_zh["sentiment_label"]
    .value_counts(dropna=False)
)

print("Sentiment Label Distribution")
print(sentiment_distribution_zh)

print("\nSentiment Label Percentage")
print(
    pilot_zh["sentiment_label"]
    .value_counts(normalize=True, dropna=False)
    .mul(100)
    .round(1)
)

Sentiment Label Distribution
sentiment_label
Negative    17
Positive    13
Name: count, dtype: int64

Sentiment Label Percentage
sentiment_label
Negative    56.7
Positive    43.3
Name: proportion, dtype: float64


In [95]:
# Confidence distribution

confidence_distribution_zh = (
    pilot_zh["confidence"]
    .value_counts(dropna=False)
)

print("Confidence Distribution")
print(confidence_distribution_zh)

print("\nConfidence Percentage")
print(
    pilot_zh["confidence"]
    .value_counts(normalize=True, dropna=False)
    .mul(100)
    .round(1)
)

Confidence Distribution
confidence
High      25
Medium     5
Name: count, dtype: int64

Confidence Percentage
confidence
High      83.3
Medium    16.7
Name: proportion, dtype: float64


In [96]:
# Extract cases requiring additional QA review

qa_cases_zh = pilot_zh[
    (pilot_zh["confidence"] == "Low") |
    (pilot_zh["sentiment_label"].isin(["Mixed", "Unclear"]))
].copy()

print("Number of QA cases:", len(qa_cases_zh))

qa_cases_zh[
    [
        "review_id",
        "review_title",
        "review_body",
        "sentiment_label",
        "confidence",
        "notes"
    ]
]

Number of QA cases: 0


,review_id,review_title,review_body,sentiment_label,confidence,notes


### Chinese Pilot QA Summary

- Total pilot samples: **30**
- Missing sentiment labels: **0**
- Missing confidence values: **0**
- Negative: **17 (56.7%)**
- Positive: **13 (43.3%)**
- Mixed: **0**
- Neutral: **0**
- Unclear: **0**
- High confidence: **25 (83.3%)**
- Medium confidence: **5 (16.7%)**
- Low confidence: **0**
- Additional QA cases: **0**

#### QA Review

No `Mixed`, `Unclear`, or `Low`-confidence cases were identified
for additional QA review.

Several borderline cases were already documented in the `notes`
field during pilot annotation, including dominant sentiment,
product-vs-fulfillment sentiment, implicit negative expressions,
and template-like review content.

No annotation changes were required after QA.

## PA-18 Save Chinese Pilot Annotation

Save the completed and QA-reviewed Chinese pilot annotations as a
separate CSV file while preserving the original 300-sample annotation
dataset for the main annotation stage.

In [97]:
# Save Chinese pilot annotation

output_path_zh = Path(
    "../data/annotation/amazon_reviews_zh_pilot_annotation_30.csv"
)

pilot_zh.to_csv(
    output_path_zh,
    index=False,
    encoding="utf-8-sig"
)

print("Saved to:", output_path_zh)
print("File exists:", output_path_zh.exists())

Saved to: ..\data\annotation\amazon_reviews_zh_pilot_annotation_30.csv
File exists: True


In [98]:
# Verify saved Chinese pilot annotation

saved_pilot_zh = pd.read_csv(output_path_zh)

print("Saved dataset shape:", saved_pilot_zh.shape)

print(
    "Missing sentiment labels:",
    saved_pilot_zh["sentiment_label"].isna().sum()
)

print(
    "Missing confidence values:",
    saved_pilot_zh["confidence"].isna().sum()
)

saved_pilot_zh.head()

Saved dataset shape: (30, 8)
Missing sentiment labels: 0
Missing confidence values: 0


,review_id,review_title,review_body,language,product_category,sentiment_label,confidence,notes
0,zh_0842626,质量实在一般,拿到手非常喜欢，轻便，简单，耐看。但是质量很p，五一假期穿着跟朋友玩了会儿篮球斗牛，结果左脚小白鞋直接坏掉：后跟脱胶，前面鞋面蹭破。此外，感觉时间一长大拇指会把鞋面顶破，质量实在不敢恭维。我穿三叶草打篮球搞运动什么的，完全不会出现一丢丢问题。,zh,shoes,Negative,High,"Positive comments on appearance and comfort, but serious quality problems dominate"
1,zh_0531516,服务态度超好！,服务态度超好，衣服质量也是杠杠的，谢谢，祝生意越做越好！,zh,apparel,Positive,High,NaN
2,zh_0474460,刑法学,书还不错 纸张也还好 发货到货都快。,zh,book,Positive,High,"Mild positive wording, but the overall positive sentiment is unambiguous"
3,zh_0915261,书可以，快递太差,周四下单时显示有货，周六送到，还特地请了假等书，结果周六凌晨发邮件给我说没货，下周一送到，然后没通知的情况下快递又周日送到，我不在，还被快递大骂一顿，说不在怎么不选工作日送货,zh,book,Negative,High,"Product itself is acceptable, but negative delivery and service experience dominates"
4,zh_0581507,好书，值得推荐！,昨天买的，今天就送到了，大致流浪了一下，里面的问题都是在教育孩子的过程中遇到的典型问题，好书，值得推荐！,zh,book,Positive,High,NaN


## PA-19 Chinese Pilot Findings & Guideline Issues

The Chinese pilot annotation identified several language-specific
and cross-language issues that should be clarified in the next
version of the annotation guideline.

### 1. Mild Chinese Sentiment Expressions

Some Chinese reviews expressed sentiment using relatively mild
phrases such as:

- 还不错
- 还好
- 可以
- 一般
- 有点

These expressions may indicate weak sentiment intensity, but they
do not necessarily imply low annotation confidence.

**Example**

> "书还不错 纸张也还好 发货到货都快。"

The positive sentiment is mild, but the overall evaluation is
consistently positive.

**Decision:** `Positive / High`

**Guideline issue:**  
Clarify that mild sentiment intensity should not automatically
reduce confidence.

---

### 2. Implicit Negative Sentiment

Several reviews expressed dissatisfaction without using strong
negative adjectives.

Negative sentiment was sometimes communicated through:

- unmet expectations,
- loss of interest,
- rhetorical questions,
- comparison with better alternatives,
- delayed shipping or service failure.

**Example**

> "以为可以媲美东野圭吾，不过是我的一厢情愿。"

The review does not rely on a strong explicit negative word,
but disappointment and unmet expectations are clear.

**Decision:** `Negative / High`

**Guideline issue:**  
Add Chinese examples of implicit negative sentiment and
context-based interpretation.

---

### 3. Dominant Sentiment in Reviews Containing Both Positive and Negative Comments

Several Chinese reviews contained both positive and negative
evaluations.

The presence of words such as:

- 但是
- 不过
- 就是

should not automatically lead to a `Mixed` label.

The annotator should determine which evaluation dominates the
overall review.

**Example**

> "质量好又便宜... 有些功能没什么用... 用久了觉得好舒服。"

Although several minor concerns are mentioned, the final product
evaluation is clearly positive.

**Decision:** `Positive / High`

**Guideline issue:**  
Add contrastive Chinese examples for:

- Positive with minor negative comments
- Negative with minor positive comments
- Mixed sentiment

---

### 4. Product Sentiment vs. Fulfillment and Service Experience

As in the English pilot, several Chinese reviews evaluated the
product positively while expressing dissatisfaction with:

- shipping,
- packaging,
- stock availability,
- customer service,
- order fulfillment.

**Example**

> "鞋是好鞋... 鞋盒都破了，鞋子也刮花了..."

The product itself is evaluated positively, while the purchasing
and delivery experience is strongly negative.

**Decision:** `Negative / High`

**Guideline issue:**  
Define a consistent sentiment target for reviews containing
multiple evaluation targets.

Possible targets include:

- product quality,
- content quality,
- delivery,
- packaging,
- seller service,
- overall customer experience.

---

### 5. Content Quality vs. Product or Edition Quality

Some book reviews distinguished between the quality of the content
and the quality of the physical or digital edition.

**Example**

> "书是很棒了... 但是中间的低级错误太多了。"

The book content is positively evaluated, while the Kindle edition
contains many formatting and text errors.

**Decision:** `Negative / Medium`

**Guideline issue:**  
Clarify how to annotate reviews where content quality and edition
or production quality have different sentiment directions.

---

### 6. Template-like or Reward-driven Reviews

One review contained positive wording but most of the text described
copying generic review text to earn reward points.

**Example**

> "很好... 这段话复制走了... 既能赚积分，还省事..."

**Decision:** `Positive / Medium`

The review contains positive wording, but actual product-specific
evaluation is limited.

**Guideline issue:**  
Add a rule for template-like, spam-like, or reward-driven reviews.

Future versions should clarify when such reviews should remain
Positive or Negative and when they should be labeled `Unclear`
because the text does not contain reliable product evaluation.

---

### 7. Chinese Pilot QA Summary

The Chinese pilot consisted of 30 reviews.

Final sentiment distribution:

- Negative: **17 (56.7%)**
- Positive: **13 (43.3%)**
- Mixed: **0**
- Neutral: **0**
- Unclear: **0**

Confidence distribution:

- High: **25 (83.3%)**
- Medium: **5 (16.7%)**
- Low: **0**

No missing sentiment labels or confidence values were found.

No `Mixed`, `Unclear`, or `Low`-confidence cases were identified
through the automatic QA filter.

However, several borderline cases were documented in the `notes`
field during manual annotation.

---

### Proposed Guideline v0.2 Updates

Based on the Chinese pilot, the next guideline revision should consider:

1. Adding examples for mild Chinese sentiment expressions.
2. Clarifying that sentiment intensity and annotation confidence are different.
3. Adding examples of implicit negative sentiment.
4. Expanding the dominant-sentiment rule for Chinese reviews.
5. Defining the sentiment target when product, delivery, packaging,
   service, and content evaluations conflict.
6. Clarifying content quality vs. edition or production quality.
7. Adding guidance for template-like or reward-driven reviews.

These findings will be combined with the Korean and English pilot
results before updating the guideline from v0.1 to v0.2.

# PA-20 Guideline v0.2 Revision Plan

The multilingual pilot annotation was conducted on Korean, English,
and Chinese review samples.

Based on the pilot findings and QA results, the following revisions
will be applied to the annotation guideline before the main
annotation stage.

## 1. Correct Label Set

The official label set is:

`Positive | Negative | Neutral | Mixed | Unclear`

`Negative` was omitted from the label-set summary in v0.1 and will
be corrected in v0.2.

---

## 2. Clarify Neutral vs. Unclear

### Neutral

Use `Neutral` when the meaning of the review is understandable,
but the text contains no meaningful positive or negative evaluation.

### Unclear

Use `Unclear` when the sentiment itself cannot be reliably
determined because the text is incomplete, ambiguous,
context-dependent, meaningless, or insufficient.

### Decision Rule

Ask:

**Can the meaning of the review be understood reliably?**

- Yes, but no positive or negative evaluation exists → `Neutral`
- No, the intended sentiment cannot be determined reliably → `Unclear`

Weak sentiment alone must not be labeled `Neutral`.

---

## 3. Expand the Dominant Sentiment Rule

When positive and negative opinions appear together, do not determine
the label by counting positive and negative expressions.

Consider:

1. Main purpose of the review
2. Importance of the evaluated feature
3. Core product or service functionality
4. Review title
5. Final conclusion or recommendation
6. Severity of the reported problem

If one sentiment clearly dominates the overall evaluation,
assign the dominant sentiment.

Use `Mixed` only when both positive and negative evaluations are
meaningful and neither clearly dominates.

The presence of contrastive expressions such as:

- EN: but / however / although
- KO: 하지만 / 그런데 / 다만
- ZH: 但是 / 不过 / 就是

does not automatically indicate `Mixed`.

---

## 4. Define the Sentiment Target

The annotation target will be the **overall customer experience
expressed in the review**, not only the physical product.

Relevant evaluation targets may include:

- Product quality
- Product functionality
- Content quality
- Digital or physical edition quality
- Delivery
- Packaging
- Order fulfillment
- Seller or customer service

When multiple targets have different sentiment directions,
apply the dominant sentiment rule.

If meaningful positive and negative evaluations remain balanced,
use `Mixed`.

---

## 5. Clarify Core Function vs. Secondary Attributes

Failure of a core product or service function may outweigh
minor positive attributes.

For example:

- Attractive design + major functional failure → usually `Negative`
- Minor packaging issue + excellent product performance → usually `Positive`

Annotators should evaluate the importance of each issue rather
than the number of positive or negative statements.

---

## 6. Clarify Confidence Level

Confidence represents:

**certainty in the annotation decision**

and does NOT represent:

**strength or intensity of sentiment**

### High

The annotator can confidently determine the appropriate label,
even when the sentiment expression itself is mild.

### Medium

Some interpretation or comparison between competing signals
is required.

### Low

The intended sentiment or appropriate label remains substantially
ambiguous or highly context-dependent.

An `Unclear` annotation may still receive `High` confidence when
the annotator is highly certain that the available text is
insufficient for sentiment determination.

---

## 7. Expand Language-Specific Rules

### Korean

Add guidance for:

- ㅋㅋ / ㅎㅎ / ㅠㅠ
- Internet slang and abbreviations
- Double negatives
- Honorific sarcasm
- Incomplete or context-poor expressions

Laughter markers alone must not determine sentiment.

### English

Add guidance for:

- Negation
- Understatement
- Sarcasm
- Intensifiers
- Contrastive structures
- Core-function failure vs. minor positive attributes

### Chinese

Add guidance for mild expressions such as:

- 还不错
- 还好
- 还行
- 一般
- 有点

Weak sentiment intensity does not automatically imply
`Neutral` or lower confidence.

Also consider implicit negative sentiment expressed through:

- Unmet expectations
- Rhetorical questions
- Comparison
- Loss of interest
- Service or delivery failure

---

## 8. Add Guidance for Template-like Reviews

Reviews that contain generic, copied, reward-driven, or
template-like text require additional attention.

If reliable sentiment toward the reviewed item or customer
experience is still present, annotate that sentiment.

If the text does not provide enough reliable evaluative
information, consider `Unclear`.

Document borderline cases in `notes`.

---

## 9. Expand Notes Requirements

`notes` are especially recommended for:

- Mixed sentiment
- Unclear cases
- Low confidence
- Sarcasm
- Language-specific expressions
- Dominant-sentiment decisions involving competing signals
- Product vs. fulfillment conflicts
- Multiple sentiment targets
- Template-like or reward-driven reviews

---

## 10. QA Rule Update

QA should include:

- All `Low` confidence cases
- All `Mixed` cases
- All `Unclear` cases
- Missing annotation fields
- Inconsistent application of dominant sentiment
- Sentiment-target conflicts
- Recurring language-specific ambiguity
- Cross-language consistency checks

Class distributions should be monitored but should not be
artificially balanced during annotation.

---

## Revision Decision

`Guideline v0.1`
→ `Multilingual Pilot Annotation`
→ `Pilot Findings & QA`
→ **`Guideline v0.2`**
→ `Main Annotation`
→ `QA & Ground Truth Comparison`
→ `Error Analysis`
→ `Guideline v1.0`